plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -553,19 +725,33 @@ class EthereumApi(LedgerApi, EthereumHelper):
 
     identifier = _ETHEREUM
 
+    _gas_price_strategies: Dict[str, Callable] = {
+        "gas_station": get_gas_price_strategy,
+        "eip1559": get_gas_price_strategy_eip1559
+    }
+
     def __init__(self, **kwargs: Any):
         """
         Initialize the Ethereum ledger APIs.
 
         :param kwargs: keyword arguments
         """
+        _default_logger.info(kwargs)
         self._api = Web3(
             HTTPProvider(endpoint_uri=kwargs.pop("address", DEFAULT_ADDRESS))
         )
         self._chain_id = kwargs.pop("chain_id", DEFAULT_CHAIN_ID)
-        self._gas_price_api_key = kwargs.pop("gas_price_api_key", None)
         self._is_gas_estimation_enabled = kwargs.pop("is_gas_estimation_enabled", False)
 
+        gas_price_strategy = kwargs.pop("gas_price_strategy", None)
+        if gas_price_strategy is None:
+            gas_price_strategy = DEFAULT_GAS_PRICE_STRATEGY.copy()
+
+        self._gas_price_strategy = gas_price_strategy.pop("strategy")
+        self._gas_price_strategy_callable = self._gas_price_strategies.get(
+            self._gas_price_strategy)
+        self._gas_price_strategy_kwargs = gas_price_strategy.pop("kwargs")
```

Comment:
@DavidMinarsch Is it really a good idea to make the gas price strategy configurable from both the constructor and method. If we need to use different strategies throughout the execution making it configurable from the method would be a good idea and If we use the same strategy throughout it will be better if we make it configurable only from the constructor.

Reply:
Since we configure the strategy params from the .yml file I propose we use following structure to configure it

```yml
config:
  ledger_apis:
    ethereum:
      address: ...
      gas_price_strategy:
        strategy: gas_station
        kwargs:
          gas_price_api_key: ...
          method: ...
```

and for eip1559

```yml
config:
  ledger_apis:
    ethereum:
      address: ...
      gas_price_strategy:
        strategy: eip1559
        kwargs:
          max_gas_fast: ...
          fee_history_blocks: ...
          fee_history_percentile: ...
          priority_fee_estimation_trigger: ...
          default_priority_fee: ...
          fallback_estimate: ...
          priority_fee_increase_boundary: ...
```

Reply:
Yes that works!

Reply:
However, this format does not support configuring both and then setting one as default...

Reply:
How about:
```
config:
  ledger_apis:
    ethereum:
      address: ...
      default_gas_price_strategy: eip1559
      gas_price_strategies:
          eip1559:
              max_gas_fast: ...
              ....
          gas_station:
              gas_price_api_key: ...
              ....
```

Reply:
This then sets the global default strategy but lets you specify configs for all of them. Then on a tx level you can select different strategy from default and even override its kwargs